src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -4,61 +4,313 @@
         {
-            // Filter out nested commands, and commands that aren't bound to a
-            // key. This page is currently just for displaying the actions that
-            // _are_ bound to keys.
-            if (command.HasNestedCommands() || !command.Keys())
+            if (senderVM.IsInEditMode())
             {
-                continue;
+                // Ensure that...
+                // 1. we move focus to the edit mode controls
+                // 2. this is the only entry that is in edit mode
+                for (uint32_t i = 0; i < _KeyBindingList.Size(); ++i)
+                {
+                    const auto& kbdVM{ _KeyBindingList.GetAt(i) };
+                    if (senderVM == kbdVM)
+                    {
+                        // This is the view model entry that went into edit mode.
+                        // Move focus to the edit mode controls by
+                        // extracting the list view item container.
+                        const auto& container{ KeyBindingsListView().ContainerFromIndex(i).try_as<ListViewItem>() };
+                        container.Focus(FocusState::Programmatic);
+                    }
+                    else
+                    {
+                        // Exit edit mode for all other containers
+                        get_self<KeyBindingViewModel>(kbdVM)->DisableEditMode();
+                    }
+                }
+
+                const auto& containerBackground{ Resources().Lookup(box_value(L"EditModeContainerBackground")).as<Windows::UI::Xaml::Media::Brush>() };
+                get_self<KeyBindingViewModel>(senderVM)->ContainerBackground(containerBackground);
+            }
+            else
+            {
+                // Focus on the list view item
+                KeyBindingsListView().ContainerFromItem(senderVM).as<Controls::Control>().Focus(FocusState::Programmatic);
+
+                const auto& containerBackground{ Resources().Lookup(box_value(L"NonEditModeContainerBackground")).as<Windows::UI::Xaml::Media::Brush>() };
+                get_self<KeyBindingViewModel>(senderVM)->ContainerBackground(containerBackground);
             }
-            keyBindingList.push_back(command);
         }
-        std::sort(begin(keyBindingList), end(keyBindingList), CommandComparator{});
-        _filteredActions = single_threaded_observable_vector<Command>(std::move(keyBindingList));
     }
 
-    Collections::IObservableVector<Command> Actions::FilteredActions()
+    void Actions::_ViewModelDeleteKeyBindingHandler(const Editor::KeyBindingViewModel& /*senderVM*/, const Control::KeyChord& keys)
     {
-        return _filteredActions;
+        // Update the settings model
+        _State.Settings().ActionMap().DeleteKeyBinding(keys);
+
+        // Find the current container in our list and remove it.
+        // This is much faster than rebuilding the entire ActionMap.
+        if (const auto index{ _GetContainerIndexByKeyChord(keys) })

```

Comment:
why can't you get the index of senderVM in the list of ViewModel objects? That would be easier in all cases than walking every key binding and checking _by key chord_...... right?

Reply:
Switched to `IndexOf`